File: LC/library/fragment-stack-manager/src/main/java/com/tieudieu/fragmentstackmanager/ScreenManager.java
package com.tieudieu.fragmentstackmanager;

import android.support.v4.app.Fragment;

/**
 * Created by chienchieu on 27/01/2016.
 */
public interface ScreenManager {

    void onMainScreenRequested();
    void onSwapFragmentRequested(Fragment fragment);
    void onBackFragmentRequested();
    void onFragmentEntered(Fragment fragment);
    void onCloseRequested();
    void onNewScreenRequested(int indexTag, int typeContent, Object object);
    void onNewScreenRequested(int indexTag, String typeContent, Object object);

}

File: LC/library/fragment-stack-manager/src/main/java/com/tieudieu/fragmentstackmanager/ScreenManager.kt
package com.tieudieu.fragmentstackmanager

import android.support.v4.app.Fragment

/**
 * Created by chienchieu on 27/01/2016.
 */
interface ScreenManager {
    fun onMainScreenRequested()
    fun onSwapFragmentRequested(fragment: Fragment)
    fun onBackFragmentRequested()
    fun onFragmentEntered(fragment: Fragment?)
    fun onCloseRequested()
    fun onNewScreenRequested(indexTag: Int, typeContent: String?, obj: Any?)
    fun onNewScreenRequested(indexTag: Int, fragment: Fragment?, obj: Any?)
}
